src/main/ruby/core/truffle/cext.rb

```diff
@@ -181,8 +181,12 @@ def fd
   end
 
   class RStringPtr
+    private_class_method :new
+
     attr_reader :string
 
+    # If you're looking to create an RStringPtr object, you should probably be using
+    # `string_get_rstring_ptr` primitive instead.
```

Comment:
It would be nice to have a comment somewhere about why we need to keep these RStringPtr unique per String.

Reply:
and also that we rely on Cext mutex to synchronize the ptr setting. Thanks.

Reply:
Sorry I've missed the widow before merge.